src/storage/meta/entry/segment_entry.cppm

```diff
@@ -175,22 +179,23 @@ private:
     const SegmentID segment_id_{};
     const SizeT row_capacity_{};
     const u64 column_count_{};
-    atomic_flag sealed_;
 
     mutable std::shared_mutex rw_locker_{}; // protect following
 
     SizeT row_count_{};
     SizeT actual_row_count_{}; // not deleted row count
 
-    TxnTimeStamp min_row_ts_{UNCOMMIT_TS};      // Indicate the commit_ts which create this SegmentEntry
+    TxnTimeStamp min_row_ts_{UNCOMMIT_TS}; // Indicate the commit_ts which create this SegmentEntry
+    TxnTimeStamp max_row_ts_{UNCOMMIT_TS};
     TxnTimeStamp first_delete_ts_{UNCOMMIT_TS}; // Indicate the first delete commit ts. If not delete, it is UNCOMMIT_TS
-    TxnTimeStamp compacting_ts_{UNCOMMIT_TS};   // Indicate the commit_ts which start compacting this SegmentEntry
-    TxnTimeStamp no_delete_ts_{UNCOMMIT_TS};    // Indicate the commit_ts which prehibit delete in this SegmentEntry
-    TxnTimeStamp deprecate_ts_{UNCOMMIT_TS};    // Indicate the commit_ts which deprecate this SegmentEntry
 
     Vector<SharedPtr<BlockEntry>> block_entries_{};
 
     CompactSegmentsTask *compact_task_{};
+    SegmentStatus status_;
+
+    std::condition_variable_any txn_num_cv_{};
```

Comment:
Why not use std::condition_variable?

Reply:
condition_variable cannot wait on std::shared_mutex (only work on std::mutex)